provider/cloudflare/cloudflare.go

```diff
@@ -263,10 +263,16 @@ func (p *CloudFlareProvider) Records(ctx context.Context) ([]*endpoint.Endpoint,
 // ApplyChanges applies a given set of changes in a given zone.
 func (p *CloudFlareProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) error {
 	cloudflareChanges := []*cloudFlareChange{}
+	cloudflarePreChanges := []*cloudFlareChange{}
```

Comment:
Not sure how this resolves an actuall issue. Is the pod killed due to null-pointer or there is incorrect SIGTERM handling?

Would you be able to share a bit more about an actual scenario and the logs right before the crash happens?

Is this still a case for 0.15.1?
/assign

Reply:
a pod can be killed without SIG signal (if the node has a power problem, s*** happened) and in this case, the update will be done only in the txt and not in the real record !

Reply:
Ok. I would assume the service could get SIGKILL or `kubectl delete --force` or similar. 

I've personally observed that Cloudflare DNS changes can take some time to propagate. This suggests that the service might not always create all necessary record types simultaneously, potentially leading to inconsistencies if something 

I did skimmed trough the code. Probably worth to use batching for records update https://pkg.go.dev/github.com/cloudflare/cloudflare-go/v4/dns#RecordService.Batch

Reply:
`RecordService.Batch` appears in v4 cloudflare lib
and this lib is only available for 4 weeks, not december 2023 !